wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -213,7 +213,7 @@ instance WalletAPI MockWallet where
         else
             -- This is the coin selection algorithm
             -- TODO: Should be customisable
-            let funds = P.takeWhile ((vl <) . snd)
+            let funds = takeUntil ((vl >=) . snd)
```

Comment:
I'm confused - this looks backwards to me. Surely we should take until the taken amount is greater than `vl`, so we have more than it? Honestly, this whole expression is pretty incomprehensible.

I also don't see how this isn't equivalent to the previous one...

Reply:
> I also don't see how this isn't equivalent to the previous one...

`takeUntil p` also gives you the first element that satisfies the predicate, and `takeWhile (not . p)` doesn't

But I take your point that it is a bit intransparent, I'll make it nicer

Reply:
Also you could document this - it's not immediately clear to me that "takeUntil" will or won't include the first element that satisfies the predicate.